### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kasimhelil**

### CODE REVIEW

**Comment by tsegay** [comment]
@kasimhelil can you add references to all the issues addressed on the PR

**Comment by kasimhelil** [comment]
This PR should resolve the following issues #1563  #1556  #1562  #1539 .
I didn't create an issue for seat number to be added to the .csv file

File: gtas-parent/gtas-query-builder/src/main/java/gov/gtas/querybuilder/service/QueryBuilderService.java [hunk 1]
```diff
@@ -211,6 +215,15 @@ public PassengersPageDto runPassengerQuery(QueryRequest queryRequest) throws Inv
 					Passenger passenger = (Passenger) result[1];
 					Flight flight = (Flight) result[2];
 					PassengerGridItemVo vo = createPassengerGridItemVo(paxDocuments, passenger, flight);
+					
+					List<Seat> seatList = seatRepository.findByFlightIdAndPassengerId(flight.getId(), passenger.getId());
+					if (CollectionUtils.isNotEmpty(seatList)) {
+						List<String> seats = seatList.stream().map(seat -> seat.getNumber()).distinct()
+								.collect(Collectors.toList());
+						if (seats.size() == 1) {
```
Comment (simbamarufu1) [inline gtas-parent/gtas-query-builder/src/main/java/gov/gtas/querybuilder/service/QueryBuilderService.java]: What could cause a passenger to have more than one seatId on a flight?

File: gtas-parent/gtas-query-builder/src/main/java/gov/gtas/querybuilder/service/QueryBuilderService.java [hunk 2]
```diff
@@ -211,6 +215,15 @@ public PassengersPageDto runPassengerQuery(QueryRequest queryRequest) throws Inv
...
+							vo.setSeat(seats.get(0));
+						}
+					}
```
Comment (tsegay) [inline gtas-parent/gtas-query-builder/src/main/java/gov/gtas/querybuilder/service/QueryBuilderService.java]: Reusable code from https://github.com/US-CBP/GTAS/blob/dev/gtas-parent/gtas-webapp/src/main/java/gov/gtas/controller/PassengerDetailsController.java#L120-L127

You might need to put this on a service and reference it from both places.

**Comment by tsegay** [comment]
https://github.com/US-CBP/GTAS/pull/1570#discussion_r352188448 That is a good question. Because the unique key for the seat table is defined based on `number`,`apis`,`passenger_id`,`flight_id` fields, It is possible a passenger could have more than one seatId. we might need to revisit that to just consider the `passenger_id`,`flight_id` for the uniqueness.

File: gtas-parent/gtas-webapp/src/main/java/gov/gtas/controller/PassengerDetailsController.java [hunk 3]
```diff
@@ -94,6 +94,9 @@
 	
 	@Autowired
 	private NoteTypeService noteTypeService;
+	
+	@Autowired
+	private SeatServiceImpl seatService;
```
Comment (tsegay) [inline gtas-parent/gtas-webapp/src/main/java/gov/gtas/controller/PassengerDetailsController.java]: you need to Inject the interface here.

File: gtas-parent/gtas-commons/src/main/java/gov/gtas/services/SeatServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import gov.gtas.model.Seat;
+import gov.gtas.repository.SeatRepository;
+
+@Service
+public class SeatServiceImpl implements SeatService{
```
Comment (tsegay) [inline gtas-parent/gtas-commons/src/main/java/gov/gtas/services/SeatServiceImpl.java]: the formatting looks off.

**Pull request merged**